From dcae1bb2071378eeade704295a4488a8edd03b2a Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sat, 18 Feb 2012 13:16:17 +0200
Subject: [PATCH 41/73] Use WPA_INTERFACE_DISABLED with interface removed
 events

This makes WPA_INTERFACE_DISABLED more consistent in indicating that
wpa_supplicant cannot currently control the interface regardless of
whether the interface is disabled or completely removed.

Signed-hostap: Jouni Malinen <j@w1.fi>
intended-for: hostap-1
(cherry picked from commit cb6710a4a5725c036b69ff751545c32d37f2e7de)
---
 wpa_supplicant/events.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -1736,11 +1736,13 @@ wpa_supplicant_event_interface_status(st
 			wpa_msg(wpa_s, MSG_INFO, "Failed to initialize the "
 				"driver after interface was added");
 		}
+		wpa_supplicant_set_state(wpa_s, WPA_DISCONNECTED);
 		break;
 	case EVENT_INTERFACE_REMOVED:
 		wpa_dbg(wpa_s, MSG_DEBUG, "Configured interface was removed");
 		wpa_s->interface_removed = 1;
 		wpa_supplicant_mark_disassoc(wpa_s);
+		wpa_supplicant_set_state(wpa_s, WPA_INTERFACE_DISABLED);
 		l2_packet_deinit(wpa_s->l2);
 		wpa_s->l2 = NULL;
 #ifdef CONFIG_IBSS_RSN
